magma_ff/create_metawfr.py

```diff
@@ -558,6 +562,42 @@ def cast_parameter_value(self, parameter_value):
         return result
 
 
+def get_files_for_file_formats(file_items, file_formats, requirements=None):
```

Comment:
How about this? 
```
def get_files_for_file_formats(file_items, file_formats, requirements=None):
    result = []

    for file_item in file_items:
        
        file_item_format = file_item.get(FILE_FORMAT, {}).get(FILE_FORMAT)

        if file_item_format in file_formats:
            file_uuid = file_item[UUID]
            if requirements:
                if all(key in file_item.keys() for key in requirements.keys()):
                    if all(file_item[key] in accepted_values for key, accepted_values in requirements.items()):
                        result.append(file_uuid)
            else:
                result.append(file_uuid)

    return result
```

I rearranged the order of checks starting from `file_item_format`. This way we can avoid the iteration over requirements if `file_item_format` is not included `file_formats` and also we don’t have this `requirements_met `variable

Reply:
Makes sense. Implemented slightly differently.